### PR TITLE
Braintree 3DSecure: store credit card

### PR DESCRIPTION
**Pull request by damianpm**
**What this PR does / why we need it**:

To comply with SCA, we need to add `threeDSecure` support for **Braintree**

Fixes: https://issues.redhat.com/browse/THREESCALE-6495
Part of SCA Epic: https://issues.redhat.com/browse/THREESCALE-6515

**TODOS:**

  - [x] Store Credit card  – (This PR)
  - [ ] Offline payment (Follow-up task)
  - [ ] Payment page (?) (Follow-up task)
  - [ ] Webhook – Probably not needed as Braintree web hooks do not include credit card payments status changes (docs)
  - [ ] Failed payment email (Follow-up task, shared task with Stripe integration)
  
 This PR handles only the first step, next steps will be handled independently

**Verification steps** 

**Update dependencies:**
```
yarn clean && yarn install
```
**Enable payments option:**
By starting rails server with flag `THREESCALE_PAYMENTS_ENABLED`:
```
UNICORN_WORKERS=8 THREESCALE_PAYMENTS_ENABLED=1 RAILS_LOG_TO_STDOUT=1 bundle exec rails server
```
Or, by setting `config.three_scale.payments.enabled = true` in `config/environments/development.rb `

**Get keys from Braintree:**
Go to [Braintree](https://www.braintreepayments.com/sandbox), sign up and get your  **Public Key, Merchant ID** and **Private Key**

**Enable charging:**
In **Admin Portal**,

- Go to **Audience > Billing > Charging & Gateway**
- Select **Charging enabled**
- Click **Save**
- In **Credit card gateway > Gateway**, select **Braintree (Blue Platform)**
- Add your **Public Key, Merchant ID** and **Private Key**
- Click on **Save changes**

**Store Credit Card:**
In **Developer Portal**,

- Login
- Go to **Settings > Credit Card Details**
- Fill the form
- You can use a test card from: https://developers.braintreepayments.com/reference/general/testing/dotnet#credit-card-numbers
- Click on **Save details**
- A modal window (Braintree injected) should appear, with options to accept or reject, accept verification.

### CODE REVIEW

File: app/helpers/payment_details_helper.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def edit_payment_details(merchant_account = site_account)
   end
 
   def merchant_countries
-    ActiveMerchant::Country::COUNTRIES.map{|c| c[:name] }
+    ActiveMerchant::Country::COUNTRIES.map{|c| [c[:name], c[:alpha2]] }
```
[the remark this answers is not in the document]
Comment (Martouta) [reply app/helpers/payment_details_helper.rb:33]: I haven't checked after this change, but just in case, check this: [THREESCALE-6529 - Fix country format/code incompatibility in stripe](https://issues.redhat.com/browse/THREESCALE-6529)

File: app/javascript/packs/braintree3DSecureCustomerForm.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    expirationDate: {
+      selector: '#customer_credit_card_expiration_date',
+      placeholder: 'MM/YY'
+    }
+  }
+}
+
+const threeDSecureParameters = {
+  amount: '00.00',
+  billingAddress: billingAdressInfo,
```
Comment (guicassolato) [inline app/javascript/packs/braintree3DSecureCustomerForm.js:48]: It could be useful to add the email of the cardholder here, as theoretically this could open for the possibility of direct communication between PSP and cardholder in case of payment transaction failure. Not sure about data protection implications of it though.

**Review by guicassolato — APPROVED** [review]
We still need to run lots of (manual) testing on this, with different credit cards (3D-S enabled/disabled, EU-issued/abroad, etc), but it LGTM for now.

We need to be extra careful about the risk of raising the rate of denials due to some extra flags that we may have activated in the JS client. This could be a possible side-effect, maybe not to cards under the scope of SCA regulation, but to other cards. On the other, we did follow Braintree's integration guidelines and, as I said before, most of such corner cases require a proper test matrix, including with some testing in prod.

**Review by hallelujah — APPROVED** [review]
Lets merge this into the first PR
